### ISSUE TITLE
Create temporary files in TMP directory.

### ISSUE THREAD

**Issue by reznikmm** [alire-project/alire#1384]
Could you change `alr` to create all temporary files in a dedicated directory? (On non-windows use `$TMPDIR` if set, else `/tmp`). I have a bug report for `ALS`. ALS spawns two `alr` concurrently (one to get project file `alr show`, another to get envs `alr printenv`). Error was:

```
'alr --non-interactive show' failed:
ERROR: Unexpected file in index: /home/pleeb/.config/alire/indexes/community/repo/index/mo/mos6502_create_led/GNAT-TEMP-000003.TMP
```

**Comment by mosteo** [comment]
Noted. Although, are you changing into that directory on purpose? `alr` uses ./alire/tmp if inside a workspace, otherwise it uses the current location.

**Comment by reznikmm** [comment]
I couldn't reproduce this error :( I doubt the user runs vscode in the index directory, so it isn't clear to me what program creates `GNAT-TEMP-000003.TMP` there.

**Comment by mosteo** [comment]
OK, I see. That file name is likely from the GNAT compiler, although we used to use the temp file facilities in `GNAT.OS_Lib` which also use these kind of names. I'll have to check. In any case, except when we want to move temp files into place within the same filesystem, it's a better default to use the temp storage of the OS.

**Comment by berkeleynerd** [comment]
I receive this error fairly consistently when using viscose on MacOS with alire and the ada lsp. I'm on the latest release version of MacOS (Ventura 13.5) running on an M2 MacBook Air using the default intel builds of the lsp plugin, alire, and toolchain. 

``` 
last_build_profile=DEVELOPMENT
toolchain.external.gprbuild=FALSE
toolchain.use.gnat=gnat_native=12.2.1
toolchain.use.gprbuild=gprbuild=21.0.2
toolchain.external.gnat=FALSE
```

**Comment by mosteo** [comment]
This should be fixed in `master` after #1398, see https://github.com/alire-project/alire/blob/986a7d71b21058876183347ae84c35665aa563ee/src/alire/alire-platforms-common.ads#L29
